src/smvlang/expr2smv.cpp

```diff
@@ -221,6 +225,115 @@ expr2smvt::resultt expr2smvt::convert_function_application(
 
 /*******************************************************************\
 
+Function: expr2smvt::convert_typecast
+
+  Inputs:
+
+ Outputs:
+
+ Purpose:
+
+\*******************************************************************/
+
+expr2smvt::resultt expr2smvt::convert_typecast(const typecast_exprt &expr)
+{
+  // typecasts can repesent a variety of functions
+  auto &src_type = expr.op().type();
+  auto &dest_type = expr.type();
+
+  if(src_type.id() == ID_unsignedbv && dest_type.id() == ID_signedbv)
+  {
+    // unsigned to signed
+    auto src_width = to_unsignedbv_type(src_type).get_width();
+    auto dest_width = to_signedbv_type(dest_type).get_width();
+
+    if(src_width == dest_width)
+    {
+      // signess change only
+      return convert_rec(smv_signed_cast_exprt{expr.op(), dest_type});
+    }
+    else if(dest_width > src_width)
+    {
+      // Signedness _and_ width change. First go to signed, then extend.
+      return convert_rec(smv_extend_exprt{
+        smv_signed_cast_exprt{expr.op(), signedbv_typet{src_width}},
+        dest_width - src_width,
+        dest_type});
```

Comment:
Can we please have a test that covers this case? I suppose the order of operations makes a difference when the most significant bit is 1.